docs/getting-started/getting-started-http.md

```diff
@@ -0,0 +1,256 @@
+
+References:
+
+- [Http library](../standard-library/http/reference)
+- [Rest library](../standard-library/rest/reference)
+- [OpenAPI 3 emitter](../standard-library/openapi3/reference)
+
+## Setup
+
+:::note
+Make sure to have installed the [editor extension](../introduction/editor/) to get syntax highlighting and intellisense.
+:::
+
+1. Make a new folder somewhere
+2. Run `npx --package=@typespec/compiler tsp init` and select the `Generic Rest API` template
+3. Run `npm install` to install dependencies
+4. Run `npx tsp compile .` to compile the initial file
+   You can either run `npx tsp compile . --watch` to automatically compile change on save or keep running the command manually after that.
+
+Resulting file structure:
+
+```
+main.tsp
+tspconfig.yaml
+package.json
+node_modules/
+tsp-output/
+  @typespec/
+    openapi3/
+      openapi.yaml
+```
+
+## Service definition and metadata
+
+A definition for a service is the namespace that contains all the operations for the service and carries top-level metadata like service name and version. TypeSpec offers the following decorators for providing this metadata, and all are optional.
+
+- `@service` - Mark a namespace as a service namespace. Takes in the following options:
+  - `title`: Name of the service
+  - `version`: Version of the service
+- `@server` - (In `TypeSpec.Http`) the host of the service. Can accept parameters.
+
+Here's an example that uses these to define a Pet Store service:
+
+```typespec
+using TypeSpec.Http;
+using TypeSpec.Rest;
+
+/**
+ * This is a sample server Petstore server.
+ */
+@service({
+  title: "Pet Store Service",
+  version: "2021-03-25",
+})
+@server("https://example.com", "Single server endpoint")
+namespace PetStore;
+```
+
+The `server` keyword can take a third parameter with parameters as necessary:
+
+```typespec
+@server("https://{region}.foo.com", "Regional endpoint", {
+  /** Region name */
+  region?: string = "westus",
+})
+```
+
+## Resources & routes
+
+Resources are operations that are grouped in a namespace. You declare such a namespace by adding the `@route` decorator to provide the path to that resource:
+
+```typespec
+@route("/pets")
+namespace Pets {
+
+}
+```
+
+Lets add a pet model in the namespace
+
+```tsp
+model Pet {
+  @minLength(100)
+  name: string;
+
+  @minValue(0)
+  @maxValue(100)
+  age: int32;
+
+  kind: "dog" | "cat" | "fish";
+}
+```
+
+To define an operation on this resource, you need to provide the HTTP verb for the route using the `@get`, `@head` `@post`, `@put`, `@patch`, or `@delete` decorators. Alternatively, you can name your operation `list`, `create`, `read`, `update`, `delete`, or `deleteAll` and the appropriate verb will be used automatically. Lets add an operation to our `Pets` resource:
```

Comment:
This was accurate at one time but is no longer. The decision was made, rightly or wrongly, to stop implying methods from operation names. The logic now is:
> If an HTTP method decorator is not specified then the default is post if there is a body and get otherwise.

And is documented [here](https://microsoft.github.io/typespec/getting-started/typespec-for-openapi-dev#path-item-object)

Reply:
will update